src/test/java/com/exasol/adapter/dialects/snowflake/SnowflakeSqlDialectTest.java

```diff
@@ -90,40 +76,41 @@ void testSupportsJdbcSchemas() {
         assertThat(this.dialect.supportsJdbcSchemas(), equalTo(StructureElementSupport.MULTIPLE));
     }
 
-    @Test
+    @Test // TODO: if 'use schemaname' is executed in the current session then it doesnt need it. question: does it
+          // store the session?
     void testRequiresCatalogQualifiedTableNames() {
         assertThat(this.dialect.requiresCatalogQualifiedTableNames(null), equalTo(false));
     }
 
-    @Test
+    @Test // TODO: if 'use dbname' is executed in the current session then it doesnt need it. question: does it store
+          // the session?
     void testRequiresSchemaQualifiedTableNames() {
-        assertThat(this.dialect.requiresSchemaQualifiedTableNames(null), equalTo(true));
+        assertThat(this.dialect.requiresSchemaQualifiedTableNames(null), equalTo(false));
     }
 
-    @Test
+    @Test // NULLS_SORTED_AT_START is default for Snowflake unless specified otherwise
     void testGetDefaultNullSorting() {
-        assertThat(this.dialect.getDefaultNullSorting(), equalTo(NullSorting.NULLS_SORTED_AT_END));
-    }
-
-    @CsvSource({ "tableName, \"tableName\"", "table123, \"table123\"", "_table, `_table`",
-            "table_name, \"table_name\"" })
-    @ParameterizedTest
-    void testApplyQuote(final String unquoted, final String quoted) {
-        assertThat(this.dialect.applyQuote(unquoted), equalTo(quoted));
+        assertThat(this.dialect.getDefaultNullSorting(), equalTo(NullSorting.NULLS_SORTED_AT_START));
     }
 
     @Test
     void testMetadataReaderClass(@Mock final Connection connectionMock) throws SQLException {
         when(this.connectionFactoryMock.getConnection()).thenReturn(connectionMock);
         assertThat(getMethodReturnViaReflection(this.dialect, "createRemoteMetadataReader"),
-                instanceOf(AthenaMetadataReader.class));
+                instanceOf(SnowflakeMetadataReader.class));
+    }
+
+    @ValueSource(strings = { "ab:\'ab\'", "a'b:'a''b'", "a''b:'a''''b'", "'ab':'''ab'''" })
```

Comment:
Use "CSVSource" instead of handling sub-strings yourself.

Reply:
Consider `NULL` case.

Reply:
> 
> 
> Use "CSVSource" instead of handling sub-strings yourself.

from the guide:
> 
You might be wondering why I did not use the CsvSource parameterization here. This is owed to the fact that the CsvSource syntax interprets single quotes as string quotes which makes this particular scenario untestable.

Is CsvSource "stable"?

Reply:
closed.